controllers/directivebreakdown_controller.go

```diff
@@ -299,11 +304,16 @@ func (r *DirectiveBreakdownReconciler) createOrUpdatePersistentStorageInstance(c
 				if err != nil {
 					return err
 				}
+			} else {
+				if psi.Spec.UserID != dbd.Spec.UserID {
+					return dwsv1alpha1.NewResourceError(fmt.Sprintf("Existing persistent storage user ID %v does not match user ID %v", psi.Spec.UserID, dbd.Spec.UserID), nil).WithUserMessage("User ID does not match existing persistent storage").WithFatal()
```

Comment:
I'm trying to add `NewResourceErrorf` in PR #70 - consider using it if I can commit ahead of you (or add it yourself)
https://github.com/NearNodeFlash/nnf-sos/blob/fdf9573394a7efbf5dd827543f403fb87235807c/api/v1alpha1/workflow_error.go#L42-L44

Reply:
Yeah I saw that. Looks awesome. I'll switch it once your PR is in.